signal/mesos.go

```diff
@@ -0,0 +1,100 @@
+package signal
+
+import (
+	"encoding/json"
+
+	//	log "github.com/Sirupsen/logrus"
+	"github.com/dcos/dcos-signal/config"
+	"github.com/segmentio/analytics-go"
+)
+
+// Complete report used by signal service, composed of all requests
+type MesosReport struct {
+	Frameworks    []Framework `json:"frameworks"`
+	CPUTotal      int         `json:"cpus_total"`
+	MemFreeBytes  int         `json:"mem_free_bytes"`
+	MemTotalBytes int         `json:"mem_total_bytes"`
+}
+
+type Framework struct {
+	Name string `json:"name"`
+}
+
+type Mesos struct {
+	Report  *MesosReport
+	URLs    []string
+	Method  string
+	Headers map[string]string
+	Track   *analytics.Track
+}
+
+func (d *Mesos) SetReport(body []byte) error {
+	if err := json.Unmarshal(body, &d.Report); err != nil {
```

Comment:
this (and the other instance) of passing a double-pointer to Unmarshal looked strange to me at first, but if it works .. OK